src/core/electrostatics_magnetostatics/mdlc_correction.cpp

```diff
@@ -70,120 +67,92 @@ inline double g1_DLC_dip(double g, double x) {
   a = g * g * g / x + 1.5 * cc2 + 1.5 * g / x3 + 0.75 / (x3 * x);
   return a;
 }
-/* ******************************************************************* */
 
 inline double g2_DLC_dip(double g, double x) {
   double a, x2;
   x2 = x * x;
   a = g * g / x + 2.0 * g / x2 + 2.0 / (x2 * x);
   return a;
 }
-/* ******************************************************************* */
-
-/* Subroutine designed to  compute Mx, My, Mz and Mtotal  */
 
+/* Compute Mx, My, Mz and Mtotal */
 double slab_dip_count_mu(double *mt, double *mx, double *my) {
-  double node_sums[3], tot_sums[3], Mz, M, My, Mx;
-
-  node_sums[0] = 0.0;
-  tot_sums[0] = 0.0;
-  node_sums[1] = 0.0;
-  tot_sums[1] = 0.0;
-  node_sums[2] = 0.0;
-  tot_sums[2] = 0.0;
+  Utils::Vector3d node_sums{};
+  Utils::Vector3d tot_sums{};
 
   for (auto const &p : local_cells.particles()) {
     if (p.p.dipm != 0.0) {
-      auto const dip = p.calc_dip();
-      node_sums[0] += dip[0];
-      node_sums[1] += dip[1];
-      node_sums[2] += dip[2];
+      node_sums += p.calc_dip();
     }
   }
 
-  MPI_Allreduce(node_sums, tot_sums, 3, MPI_DOUBLE, MPI_SUM, comm_cart);
+  MPI_Allreduce(node_sums.data(), tot_sums.data(), 3, MPI_DOUBLE, MPI_SUM,
```

Comment:
Just as a general note: There is a return value version of `MPI_Allreduce` in boost.mpi, which can be used like
`auto const tot_sums = boost::mpi::all_reduce(comm_cart, nude_sums, std::plus<>{})`, which allows the sum
to be `const`. Also in general I think boost.mpi should be preferred, because the types and length are deduced,
which is less noisy and leave fewer places for mistakes.